orchestrator/src/test/kotlin/OrchestratorTest.kt

```diff
@@ -380,8 +381,12 @@ class OrchestratorTest : WordSpec() {
 
                 verify(exactly = 1) {
                     ortRunRepository.update(
-                        id = withArg { it shouldBe advisorJob.ortRunId },
-                        status = withArg { it.verifyOptionalValue(OrtRunStatus.FAILED) }
+                        id = withArg { it shouldBe configWorkerError.ortRunId },
+                        status = withArg { it.verifyOptionalValue(OrtRunStatus.FAILED) },
+                        issues = withArg {
+                            val issues = it.shouldBeTypeOf<OptionalValue.Present<List<Issue>>>().value
+                            issues.size shouldBe 1
```

Comment:
It would be good to check some properties of the issue itself, maybe the severity and the source. To not repeat this over and over, would it be possible to extract this logic to a helper function?

Reply:
Hmm, the helper function for verifying these properties is specific to worker error issues and I couldn't make it completely generic. What about something like this, tailored to worker error issues:

```
fun verifyIssues(
    issues: OptionalValue<Collection<Issue>>,
    validate: Collection<Issue>.() -> Unit
) {
    val issuesList = issues.shouldBeTypeOf<OptionalValue.Present<Collection<Issue>>>().value
    issuesList.validate()
}

fun Collection<Issue>.verifyWorkerErrorIssues(severity: String, source: String) {
    shouldBeSingleton { issue ->
        issue.severity shouldBe severity
        issue.source shouldBe source
    }
}

```

Reply:
Yes, that's okay. I did not mean a fully generic verification function for issues. Just one for this specific test class that covers the duplicate lines and checks some more properties.